cmd/juju/waitfor/application.go

```diff
@@ -38,13 +38,25 @@ func newApplicationCommand() cmd.Command {
 }
 
 const applicationCommandDoc = `
-Waits for an application to reach a specified state.
-arguments:
-name
-   application name identifier
-options:
---query (= 'life=="alive" && status=="active"')
-   query represents the sought state of the specified application
+The wait-for application command waits for the application to reach a goal
+state. The goal state can be defined programmatically using the query DSL
+(domain specific language). The default query for an application just waits
```

Comment:
Here and in all the other files: Which query DSL? (From our chats before, I know it's our own. I feel like we should document it somewhere.)

Reply:
Here and in all the other files: We should include that the wait-for commands are an alternative to `juju status` -- an alternative that's more targeted and computationally cheaper.

Reply:
> Here and in all the other files: Which query DSL? (From our chats before, I know it's our own. I feel like we should document it somewhere.)

That's part II, I think a discourse post should be enough.